### ISSUE TITLE
Adding spring-test-context-cache-limiter prevents cache to be reused because of different contextCustomizers instances

### ISSUE THREAD

**Issue by php-coder** [antoinemeyer/spring-test-context-cache-limiter#4]
I tried to add spring-test-context-cache-limiter to my project and I was surprised that it actually led to the cache invalidation because `contextCustomizers` had always different instance:
```
@@ -8,8 +8,8 @@
 logging.level.root=WARN
 logging.level.ru.mystamps=WARN
 org.springframework.boot.test.autoconfigure.jdbc.JdbcTestContextBootstrapper=true}'
-contextCustomizers = set[com.teketik.test.LimitingContextCustomizer@4899799b
-[ImportsContextCustomizer@fd5689d key = [org.springframework.boot.autoconfigure.cache.CacheAutoConfiguration
+contextCustomizers = set[com.teketik.test.LimitingContextCustomizer@3a5ecce3
+[ImportsContextCustomizer@aa752bb key = [org.springframework.boot.autoconfigure.cache.CacheAutoConfiguration
 org.springframework.boot.autoconfigure.flyway.FlywayAutoConfiguration
 org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
 org.springframework.boot.autoconfigure.jdbc.DataSourceTransactionManagerAutoConfiguration
```

Here are examples of test classes (they use `@JdbcTest` with identical `@ContextConfiguration`):
- [JdbcCountryDaoTest](https://github.com/php-coder/mystamps/blob/2dc3eed15969094dbf169aab100b4acbf63ace5f/src/test/java/ru/mystamps/web/feature/country/JdbcCountryDaoTest.java)
- [JdbcCategoryDaoTest](https://github.com/php-coder/mystamps/blob/2dc3eed15969094dbf169aab100b4acbf63ace5f/src/test/java/ru/mystamps/web/feature/category/JdbcCategoryDaoTest.java)

I'm using (very old) Spring Boot 2.1.18.RELEASE but it shouldn't matter as README says that "Spring Boot Test Context Cache Limiter works on any Spring Boot 2 version"

---
If you want to debug this case further, here is how to reproduce it:
- `git clone https://github.com/php-coder/mystamps`
- `cd mystamps`
- `git checkout 2dc3eed15969094dbf169aab100b4acbf63ace5f`
- `./mvnw test` -- the tests passed as the only one context is created (<sup>1</sup>)
- `vim pom.xml`
```diff
--- pom.xml
+++ pom.xml
@@ -398,6 +398,13 @@
                                </exclusion>
                        </exclusions>
                </dependency>
+
+               <dependency>
+                       <groupId>com.teketik</groupId>
+                       <artifactId>spring-test-context-cache-limiter</artifactId>
+                       <version>boot2-v1.1</version>
+                       <scope>test</scope>
+               </dependency>

                <dependency>
                        <groupId>io.qala.datagen</groupId>
@@ -1009,6 +1016,13 @@
                                <configuration>
                                        <skipTests>${skipUnitTests}</skipTests>
                                        <disableXmlReport>${disableXmlReport}</disableXmlReport>
+                                       <systemPropertyVariables>
+                                               <!--
+                                                       Specify the maximum number of contexts that should be created during unit tests
+                                                       See for details: https://github.com/antoinemeyer/spring-test-context-cache-limiter
+                                               -->
+                                               <spring.test.context.limitSize>1</spring.test.context.limitSize>
+                                       </systemPropertyVariables>
                                </configuration>
                        </plugin>

```
- `./mvnw test` -- the tests failing as now there 2 contexts are created

--
<sup>1</sup>You can ensure that there is only one context gets created by the following evidence:
1) Liquibase is running all migration only once
2) time elapsed by `JdbcCountryDaoTest` is 0.154 s while `JdbcCategoryDaoTest` took 3.717 s
3) you can also [enable debugging of cache usage by editing `src/test/resources/logback-test.xml`](https://github.com/php-coder/mystamps/commit/2dc3eed15969094dbf169aab100b4acbf63ace5f)

**Comment by antoinemeyer** [comment]
Thanks for reporting the issue.
This should be fixed by version `boot2-v1.2` (available on maven central).
Please let me know if you still have an issue!

**Comment by php-coder** [comment]
Thanks! It's working now.
